### PR TITLE
BIP-0001: Updates

### PR DESCRIPTION
**Pull request by btcdrak**
Clarified about BIP number assignment (not to self-assign, how numbers get assigned specifically).

### CODE REVIEW

File: bip-0001.mediawiki [hunk 1]
```diff
@@ -81,6 +81,8 @@ Each BIP should have the following parts:
 
 * Reference Implementation -- The reference implementation must be completed before any BIP is given status "Final", but it need not be completed before the BIP is accepted. It is better to finish the specification and rationale first and reach consensus on it before writing code.
 
+* Copyright notice -- BIPs must contain a copyright notice placing them in the public domain.
```
Comment (luke-jr) [inline bip-0001.mediawiki]: We shouldn't require (nor encourage) public domain, as some jurisdictions don't recognise it as valid...

File: bip-0001.mediawiki [hunk 2]
```diff
@@ -149,21 +153,23 @@ A BIP editor must subscribe to the Bitcoin development mailing list. All BIP-rel
 
 For each new BIP that comes in an editor does the following:
 
-* Read the BIP to check if it is ready: sound and complete. The ideas must make technical sense, even if they don't seem likely to be accepted.
+* Read the BIP to check if it is ready: sound and complete. The ideas must make technical sense, even if they don't seem likely to be accepted and must be relevent to Bitcoin.
```
Comment (luke-jr) [inline bip-0001.mediawiki]: This clause don't fit where you put it...

File: bip-0001.mediawiki (hunk 1 above)
Comment (btcdrak) [inline bip-0001.mediawiki]: If you dont require public domain then you have to add an additional requirement of permissive licensing.

File: bip-0001.mediawiki [hunk 3]
```diff
@@ -149,21 +153,23 @@ A BIP editor must subscribe to the Bitcoin development mailing list. All BIP-rel
...
-Once the BIP is ready for the repository, the BIP editor will:
+Once the BIP is ready for the repository it should be submitted as a "pull request" to the [https://github.com/bitcoin/bips bitcoin/bips] repository on GitHub where it may get further feedback.
+
+The BIP editor will:
 
-* Assign a BIP number (almost always just the next available number, but sometimes it's a special/joke number, like 666 or 3141).
+* Assign a BIP number (almost always just the next available number, but sometimes it's a special/joke number, like 666 or 3141) in the pull request comments.
 
-* Add the BIP to the [https://github.com/bitcoin/bips bitcoin/bips] repository on GitHub.
+* Merge the pull request when the author is ready (allowing some time for further peer review).
```
Comment (luke-jr) [inline bip-0001.mediawiki]: Review was supposed to occur on the Bitcoin-Development ML before being submitted as a PR. Once it's a PR, it should be ready for merging immediately (pending author ACK, if submitted by non-author).

If we want to encourage PR discussion, adding a flag like [WIP] makes sense.

File: bip-0001.mediawiki (hunk 1 above)
Comment (luke-jr) [inline bip-0001.mediawiki]: Such a requirement is already in BIP 1.

**Comment by luke-jr** [comment]
There's a lot of changes here. I suggest starting a new BIP intended to supercede BIP 1.

File: bip-0001.mediawiki (hunk 3 above)
Comment (btcdrak) [inline bip-0001.mediawiki]: I agree that's the ideal (to be ready for immediate merging), but in practice the meat gets fleshed out on the mailing list and then further peer review occurs in the PR. People are not looking for nits on the list, nor is it easy to track small nits on the mailing list. Github makes collaborative polishing a thing and we should imo, make use of it. Put another way, people are used to polishing in pull requests.

File: bip-0001.mediawiki (hunk 1 above)
Comment (btcdrak) [inline bip-0001.mediawiki]: Ah yes, I missed that.

BIPs maintainers should not merge any BIP pull-requests unless that requirement is met.

**Comment by btcdrak** [comment]
There aren't lots of changes! We're simply emphasising the don't self assign numbers rule (which needs to be written down because it's a constant problem for @gmaxwell), and changing how numbers are assigned (not by email but by requesting in a the pull, because it's more efficient and auditable).

The point about WIP is totally optional because it is a departure from the norm not to have some final polishing peer review in pull requests. It's more of a common sense addition, not a change in rules.

**Comment by sipa** [comment]
I think this should be a different BIP that overrides BIP 1. We shouldn't
update old BIPs because we want their meaning to change (beyond extra
clarification, bugfixes, examples, references, ...).

**Comment by btcdrak** [comment]
Are we seriously going to require a new BIP for a small clarification of details? Especially BIP-0001 is just a ruleset for how to manage BIPs in the first place.

**Comment by sipa** [comment]
It's a change to the semantics. I don't see why we should change an existing BIP for that.

**Comment by btcdrak** [comment]
The first BIP are the operating procedures for BIPs, unless there was some substantial change, like to the workflow (as I suggested on the ML), I really dont understand how the changes in this PR warrant a new BIP. It's incredibly wasteful - we're not writing legal statues...

We've been operating with "dont self assign numbers" for as long as I can remember. how is that a change of semantics? @gmaxwell has been plagued by people self assigning numbers, it needs to be documented surely: and documented or not, those are the rules we've been operating under.

Whether BIPs are assigned by pestering the BIP maintainer over email/IRC is requested in the body of the pull request is also such a small change it seems bizarre. Additionally, ever since we moved to Github, people have used PRs as a was to do final review (usually smaller nit, but sometimes larger ones). It works because it's more collaborative than the mailing list. Again it's not a change, it's how we have been operating, I am simply documenting the status quo.

**Comment by btcdrak** [comment]
Here's another example of something that went away when we migrated from the wiki to github

"If the BIP isn't ready, the editor will send it back to the author for revision, with specific instructions."

This isn't happening anymore. Authors write their BIPs, discuss on ML/IRC/privately and submit a PR when they have had enough feedback and feel it's ready. Any nits are dealt with in the pull request.

**Comment by btcdrak** [comment]
I have removed superfluous changes. How about this?

**Comment by maraoz** [comment]
ACK

**Comment by laanwj** [comment]
@sipa I agree with that in general, but for the meta-bip I think it makes practical sense to make an exception there:
- People will go to BIP1 to learn about writing BIPs. Having it self-contained is a plus
- There are no implementations of BIP1 - unless you are sneaky and count the other BIPs, but that'd only matter if there are big structural changes which would 'invalidate' the other BIPs by BIP1

File: bip-0001.mediawiki [hunk 4]
```diff
@@ -25,6 +25,8 @@ There are three kinds of BIP:
 
 The BIP editors assign BIP numbers and change their status. Please send all BIP-related email to the BIP editor, which is listed under [[#BIP_Editors|BIP Editors]] below. Also see [[#BIP_Editor_Responsibilities__Workflow|BIP Editor Responsibilities & Workflow]].
 
+Authors MUST NOT self assign numbers, but may use aliases such as "bip-exampleproposal".
```
Comment (andychase) [inline bip-0001.mediawiki]: I have a feeling if someone is reading this BIP they are not the type of person that self-assigns.

The readme is the best place for this notice and it's already there.

Comment (andychase) [inline bip-0001.mediawiki]: Or not? I thought there was a notice in the readme but I guess I just imagined it.

In that case I believe the notice should go in the readme after the numbers in bold. That's where people look to see that's free so you [catch them just in time](http://blog.codinghorror.com/the-just-in-time-theory/).

**Comment by andychase** [comment]
@sipa @luke-jr 

This BIP actually lists the criterion for replace vs edit right inside it:

> BIPs can also be superseded by a different BIP, rendering the original obsolete. This is intended for Informational BIPs, where version 2 of an API can replace version 1.

If we completely revamped the BIP process it might make sense to allocate a new number so the change can be advertised more clearly but in this case the minor change in process should be edited in place to prevent confusion and fragmentation.

Also a "change log" should be added at the top like [is done in BIP-0032](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) to preserve history.

File: bip-0001.mediawiki (hunk 4 above)
Comment (gmaxwell) [inline bip-0001.mediawiki]: can the example  be changed to bip-johndoe-infinitebitcoins or the like? (I prefer to include the name as duplicate work sometimes gets merged as part of the pre-assignment)

File: bip-0001.mediawiki (hunk 3 above)
Comment (andre-amorim) [inline bip-0001.mediawiki]: PR ? who come up with that such reduction ?

**Pull request merged**